shell/lisa_shell

```diff
@@ -219,6 +219,28 @@ function _lisa-install-system-check {
     _lisa-python-package-check ensurepip python3-venv
 }
 
+function _lisa-copy-git-hook {
+     # works even with worktrees
+     local hooks_dir="$(git -C "$LISA_HOME" rev-parse --git-common-dir)/hooks"
+     local hook="$1"
+
+     # Copy/Update lisa specific git-hook
+     cp "$LISA_HOME/tools/$hook.lisa" "$hooks_dir/$hook.lisa"
+
+     # Avoid overriding user hooks, but make sure lisa specific hook is
+     # called too
+     if [[ ! -e "$hooks_dir/$hook" ]]; then
+         cp "$LISA_HOME/tools/$hook" "$hooks_dir/$hook"
+     else
+         grep -qxF '"$0".lisa' "$hooks_dir/$hook" ||  echo '"$0".lisa' >> "$hooks_dir/$hook"
```

Comment:
I'm not sure there is any reliable way of making that work. If the hook ends with `exit` or something like that it will never be called.
Also, hooks can be symlinks, which means we have no idea of what we are modifying. This could be a script stored in user's home, in which case they would be pretty upset if lisa started adding random stuff to it. 

Ultimately, if someone dropped a hook in there, it's because they want that and if they want to call our things as well they can easily call it themselves in a way that makes sense. By default, no file with that name will exist since recent versions of git store the examples as `xxxx.sample` (rather than `xxxx` without the executable permission as it used to do), and most people don't have custom hooks anyway so the vast majority of users will get the feature.

Reply:
TBH I think this is too cautious. It'll help the majority of the users to not trip over an essential way of the workflow. Apps modify .bashrc all the time too to add and remove stuff, I don't see this being a problem here. It can be smarter, but I don't think it's worth the effort.

Keep in mind this only applies to their checkout lisa only. It's not a global git-hook for all projects.

By the way, if you ever use git review it automatically installs git-hooks on your behalf, which could run into the same issue.

I have no better alternative anyway.

Reply:
As mentioned, the alternative is to copy the file if the target does not exist, otherwise do nothing. The vast majority of people won't have any hook installed so it will work. People that have a hook installed will not want anyone to go and break that hook, so we don't.

> Apps modify .bashrc all the time too to add and remove stuff
That's exactly the kind of thing that makes me ditch crappy broken installers. You end up with hidden breakages waiting to explode, and you don't even have a guarantee it will work. I've spent enough time getting people to cleanup their .bashrc from PYTHONPATH setup done by lisa legacy after spending an afternoon debugging what's going on to not want that to history to be repeated :)

> By the way, if you ever use git review it automatically installs git-hooks on your behalf, which could run into the same issue.

Looking at the sources, it seems to be doing exactly what I'm describing, i.e. it won't touch anything if it detects a hook is already there:
https://opendev.org/opendev/git-review/src/branch/master/git_review/cmd.py#L1652
(the `os.access(hook_file, os.X_OK)` is there because old versions of git used to ship sample hooks with the normal filename, but without the executable bit. This means that if exec bit is missing, that is the default sample hook that can safely be overwritten).

Reply:
Hmm okay I think this makes more sense, you're right. Let me print a warning instead then.